woocommerce/woocommerce-ios
# Use 'cocoapods-repo-update' CocoaPods plugin

I have put together a CocoaPods plugin called `cocoapods-repo-update` here: https://github.com/wordpress-mobile/cocoapods-repo-update.

The purpose of this is to check if the local specs repo needs to be updated when `pod install` is run. This is particularly important for us because BuddyBuild does not provide a way of updating the repo.

This change has already been been made to WordPress iOS here: https://github.com/wordpress-mobile/WordPress-iOS/pull/10447

**Note:** Since this change relies on the `cocoapods-repo-update` gem, `pod install` now requires being run with `bundle exec`. If you prefer not to run `bundle exec pod install`, you can install the plugin like this: `gem install cocoapods-repo-update`.

---------------

To test:

- Run `bundle install`
- Run `bundle exec pod install`
- You should see `Not updating local specs repo` in the output.

To test the specs update behaviour, find a newly released pod in https://github.com/CocoaPods/Specs/commits/master and add it to the `Podfile` (I used `pod 'Backpack', '2.3.1'`).

- Run `bundle exec pod install` again.
- You should see `At least one Pod is not in the local specs repo. Updating specs repo...` in the output and it should install the new pod successfully.
